conan/tools/microsoft/visual.py

```diff
@@ -142,15 +143,31 @@ def generate(self, scope="build"):
                                 winsdk_version=winsdk_version, vcvars_ver=vcvars_ver,
                                 vs_install_path=vs_install_path)
 
-        content = textwrap.dedent("""\
+        content = textwrap.dedent(f"""\
             @echo off
             set __VSCMD_ARG_NO_LOGO=1
             set VSCMD_SKIP_SENDTELEMETRY=1
-            echo conanvcvars.bat: Activating environment Visual Studio {} - {} - winsdk_version={} - vcvars_ver={}
-            {}
-            """.format(vs_version, vcvarsarch, winsdk_version, vcvars_ver, vcvars))
+            echo conanvcvars.bat: Activating environment Visual Studio {vs_version} - {vcvarsarch} - winsdk_version={winsdk_version} - vcvars_ver={vcvars_ver}
+            {vcvars}
+            """)
         from conan.tools.env.environment import create_env_script
-        create_env_script(conanfile, content, CONAN_VCVARS_FILE, scope)
+        is_ps1 = conanfile.conf.get("tools.env.virtualenv:powershell", check_type=bool, default=False)
+        if is_ps1:
+            generator_path = conanfile.generators_path
+            content_ps1 = textwrap.dedent(f"""\
+            pushd "{generator_path}"
+            cmd /c "conanvcvars.bat&set" |
```

Comment:
we could use `$PSScriptRoot/conanvcvars.bat` instead of directly by name? this would mean we don't have to do `pushd` and `popd` - and use the same structure as `conanbuild.ps1` already does - unless I'm missing something?

(see [docs](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-7.4#psscriptroot))